lld/COFF/Options.td

```diff
@@ -299,6 +299,8 @@ def : Flag<["--"], "time-trace">, Alias<time_trace_eq>,
 def time_trace_granularity_eq: Joined<["--"], "time-trace-granularity=">,
     HelpText<"Minimum time granularity (in microseconds) traced by time profiler">;
 
+def build_id: F<"build-id">, HelpText<"Generate build ID">;
```

Comment:
I think it'd be useful if we could have a negative form of this option as well, e.g. `-build-id:none` or something like that?

For the mingw case, this option is enabled by default whenever we're linking some sort of debug info - but we might want to disable it as well.

See https://github.com/mstorsjo/llvm-mingw/issues/352 for a user request for that feature.

Reply:
I'm thinking maybe build id should be off by default for `/lldmingw` + `/debug:dwarf` as now we have the `/build-id` flag to enable it explicitly. What do you think?

Reply:
I think we could do that as well; I think it can be ok to change the MinGW specifics in the lld-link level interface, as that interface shouldn't really be used much externally.

We probably would want to keep the behaviour on the MinGW driver level the same, but we could make the MinGW driver pass the `/build-id` flag by default, unless a MinGW level `--build-id=none` is passed.

Reply:
Updated to always add `/build-id` by default from MinGW driver, and add `/build-id:no` if `-no-build-id` is given.